packages/headless/src/features/folding/folding-utils.ts

```diff
@@ -9,8 +9,13 @@ export function getAllIncludedResultsFrom(relevantResult: ResultWithFolding) {
     .map((result) => result.parentResult!);
 
   const resultsInCollection = removeDuplicates(
-    [relevantResult, ...foldedResults, ...parentResults],
-    (result) => result.uniqueId
+    [
+      ...foldedResults,
+      ...parentResults,
+      {...relevantResult, parentResult: null},
```

Comment:
As discussed live, some questions remains regarding setting `parentResult` to null. Might be hiding some other bugs...

Reply:
I made the adjustement to return the "complete" relevantResult (i.e., not remove the parentResult from it), and also fixed the failing test (by specifying distinct uniqueId values for the parent and child in the mock test result).

Looking deeper into this, I can't find any reason for this to cause any issues.